p2p/service.py

```diff
@@ -22,7 +23,10 @@ class BaseService(ABC, CancellableMixin):
     # Number of seconds cancel() will wait for run() to finish.
     _wait_until_finished_timeout = 5
 
-    def __init__(self, token: CancelToken=None) -> None:
+    # the custom event loop to run in, or None if the default loop should be used
+    target_loop: asyncio.AbstractEventLoop = None
```

Comment:
Thoughts on just naming this `loop`?  It's pretty common across all of the `asyncio` libraries for this to just be named `loop`